Update Consul secrets features docs, api-docs for 1.11 (#15854)
hashicorp/vault: website/content/api-docs/secret/consul.mdx

This endpoint creates or updates the Consul role definition. If the role does
not exist, it will be created. If the role already exists, it will receive
updated attributes. At least one of either `consul_policies`, `consul_roles`,
`service_identities`, or `node_identities` are required depending on the
Consul version.

| Method | Path                  |
| :----- | :-------------------- |
| `POST` | `/consul/roles/:name` |

### Parameters for Consul versions 1.11 and above

> - `partition` `(string: "")` - Specifies the Consul admin partition the token will be

Reviewer: Suggestion:
Specifies the Consul admin partition in which the token is generated within.

> - `partition` `(string: "")` - Specifies the Consul admin partition the token will be

Reviewer: ```suggestion
- `partition` `(string: "")` - Specifies the Consul admin partition in which the token is generated within.
```

Reviewer: You can also say:  Specifies the Consul admin partition in which the token is generated from within. 

It may be clearer if we add the word 'from' to precede the word 'within'.

> - `partition` `(string: "")` - Specifies the Consul admin partition in which the token is generated within.

Reviewer 2: "In which" means "where" so it already specifies location. Ending the sentence in "within" is redundant and awkward. The proper usage should be one of the following:
- Specifies the Consul admin partition **in which** the token is generated.
- Specifies the Consul admin partition **which** the token is generated **within**.

Personally, I prefer the first.

See
- https://linguaholic.com/linguablog/how-to-use-in-which-in-a-sentence/
- https://www.usingenglish.com/forum/threads/in-which.142245/#post-722936

Reviewer 2: I think we should also add "default" to the type info:
```suggestion
- `partition` `(string: "default")` - Specifies the Consul admin partition in which the token is generated.
```

Author: Good catch!

>   generated within. The partition must exist, and the Consul policies or roles assigned to the

Reviewer: ```suggestion
 The partition must exist, and the Consul policies or roles assigned to the
```

Reviewer: Same comment as above.

You can also say: Specifies the Consul admin partition in which the token is generated from within.

It may be clearer if we add the word 'from' to precede the word 'within'.

>   Vault role must also exist inside the given partition. If not provided, the partition `default`

Reviewer: ```suggestion
  Vault role must also exist inside the given partition. If it's not provided, the partition `default`
```

### Parameters for Consul versions 1.8 and above

> - `consul_namespace` `(string: "")` - Specifies the Consul namespace the token

Reviewer: Suggestion:
Specifies the Consul admin partition in which the token is generated within.

> - `consul_namespace` `(string: "")` - Specifies the Consul namespace the token

Reviewer: Specifies the Consul admin partition in which the token is generated within.

> - `consul_namespace` `(string: "")` - Specifies the Consul namespace the token

Reviewer: ```suggestion
- `consul_namespace` `(string: "")` - Specifies the Consul admin partition in which the token is generated within.
```

> - `consul_namespace` `(string: "")` - Specifies the Consul namespace in which the token is generated within.

Reviewer 2: Also, same comment as before to remove "within"

>   will be generated within. The namespace must exist, and the Consul policies or roles assigned

Reviewer: ```suggestion
The namespace must exist, and the Consul policies or roles assigned
```

  to the Vault role must also exist inside the given Consul namespace. If not provided, the 
  namespace `default` is used.

> - `node_identities` `(string: "")` - The list of node identities to assign to the generated

Author: Hm, I'm not sure. In the code the FieldSchema type is `TypeStringSlice` so that it builds the slice when provided multiple times via CLI.  I didn't try passing in a list via CLI to test that, does it work?

> - `node_identities` `(string: "")` - The list of node identities to assign to the generated

Reviewer 2: For the JSON list I think you would do something like

```
$ cat node_identities.json | vault write -f  foo/consul/roles/bar -
```

> - `node_identities` `(string: "")` - The list of node identities to assign to the generated

Author: Sounds good, and that also makes more sense.
